File: java/src/sample/replicant/sample/MockWidgetRepository.java
// Copyright 2009 Kiel Hodges
package replicant.sample;

import replicant.*;
import static replicant.Replicant.*;

public class MockWidgetRepository implements WidgetRepository {

  
  public Widget findWidget(int widgetId) {
    return findWidget.call(widgetId);
  }
  
  public final MethodMocker1<Widget, Integer> findWidget = 
    methodMocker(this, "findWidget").mappingResponses().with1Arg();
  
  
  
  public void storeWidget(Widget widget) {
    storeWidget.call(widget);
  }

  public final MethodMocker1<Void, Widget> storeWidget = 
    voidMethodMocker(this, "storeWidget").enforcingExpectationsStrictly().mappingResponses().with1Arg();
  
}

File: scala/src/test/scala/sample/MockWidgetRepository.scala
// Copyright 2011 Kiel Hodges
package sample

import replicant._

class MockWidgetRepository extends MockGenericRepository[Widget] { self =>

  override val mock: WidgetRepository = new BaseSubject with WidgetRepository {
    def findByPartNumber(partNumber: String): Widget = self.findByPartNumber(partNumber)
  }
    
  val findByPartNumber = method("findByPartNumber", mock.findByPartNumber _)

}
